File: src/Applications/Orbital.Web/Orbital.Forecast.BlazorApplication/Orbital.Forecast.BlazorApplication.Client/Lib/DynamicLinkLibraries/DataPerformer/DataPerformer.Portable/Helpers/TimeMeasurementProvider.cs
using System;


using DataPerformer.Interfaces;
using DataPerformer.Portable.Measurements;

namespace DataPerformer.Portable.Helpers
{
    /// <summary>
    /// Simplest provider of time measurement
    /// </summary>
    public class TimeMeasurementProvider : ITimeMeasurementProvider
    {
        #region Fields

        double time;

        const double a = 0;

        double step;

        IMeasurement timeMeasurement;

        #endregion

        #region Ctor

        /// <summary>
        /// Default constructor
        /// </summary>
        public TimeMeasurementProvider()
        {
            timeMeasurement = new TimeMeasurement(GetTime);
        }

        #endregion

        #region ITimeMeasurementProvider Members

        IMeasurement ITimeMeasurementProvider.TimeMeasurement
        {
            get
            {
                return timeMeasurement;
            }
        }

        double ITimeMeasurementProvider.Time
        {
            get
            {
                return time;
            }
            set
            {
                time = value;
            }
        }

        double ITimeMeasurementProvider.Step
        {
            get
            {
                return step;
            }
            set
            {
                step = value;
            }
        }

        #endregion

        object GetTime()
        {
            return time;
        }
    }
}

File: src/Web/reactprojectTreeEtalon/src/Library/Measurements/TimeMeasurementProvider.ts
import type { IMeasurement } from "./Interfaces/IMeasurement";
import type { ITimeMeasurementProvider } from "./Interfaces/ITimeMeasurementProvider";

export class TimeMeasurementProvider implements ITimeMeasurementProvider, IMeasurement
{
    getMeasurementName(): string
    {
        return "Time";
    }

    getMeasurementType()
    {
        return 0;
    }

    getMeasurementValue()
    {
        return this.time;
    }

    getTimeMeasurement(): IMeasurement
    {
        return this;
    }

    setTime(time: number): void
    {
        this.time = time;
    }

    getStep(): number
    {
        return 0;
    }

    setStep(time: number): void
    {
        this.step = time
    }

    getTime() : any
    {
        return this.time;
    }

    time: number = 0;

    step: number = 0
}